Side Nav UI Changes

MrRobotjs opened the pull request:
This changes the side nav look and feel and changes a lot of css under the hood to avoid alignment issues on different viewports. 

# Changed
- Changed: Changed the overall look and feel of the preference side nav and main side nav

# Fixed
- Fixed: Fixed the bulk actions bar laying out in the wrong position

# Theme
- Added: --side-nav-bar-width variable to control side nav bar width
- Added: Side nav selected tab now uses --side-nav-item-active-text-color on the text instead of just being white

![Main Page Changes](https://github.com/user-attachments/assets/b88e7514-b075-4339-81a5-52af976e06b8)
![Settings Page Side Bar Redesign](https://github.com/user-attachments/assets/d69b6326-3930-493e-964d-003c1041999e)

MrRobotjs committed:
Nav bar & search bar redesign

Rounded search bar and changed background color to white when focused
Changed search bar width when focused
Changed size of brand icon from 28px to 24px
Removed !searchFocused to stop it from deleting the activity, settings, and profile icons since the redesign of the search bar made it uneeded

MrRobotjs committed:
Changed side nav bar & preference bar width

Added --side-nav-width variable to control with ease
Removed unnecessary background-color to preference side nav bar

MrRobotjs committed:
Side nav items design improvements

Selected tab will now have the --side-nav-item-active-text-color on the text instead of just being white
Changed width of the active tab highlight bar
Side nav text & card actions will now be display: none when side nav is closed to keep icons centered
Increased text size from 18px to 21.33px
Increased tabs height
Increased tab icons width from 30px to 56px

MrRobotjs committed:
fixed preference side nav bar

Needed to redesign the side bar for the preferences page since the side bar for the main change was affecting it
Lowered width to 238px and is not attached to the wide of the main page side bar
Changed the active tab highlight bar into a circle
Fixed general spacing overall

therobbiedavis commented:
I still need to test these all across viewports and review the changes in detail, but generally speaking I'm ok with the look of this. One change I'm not in favor of is the increased border radius on the search input. It doesn't fit with the overall UX.

therobbiedavis commented on UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss:
```diff
@@ -1,5 +1,5 @@
 form {
-    max-height: 38px;
+    /*max-height: 38px;*/
```
Unless unsure if needed elsewhere, do not comment out code just go ahead and remove it.

MrRobotjs replied:
Understood, removing it

therobbiedavis commented on UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss:
```diff
@@ -17,7 +17,7 @@ form {
 .typeahead-input {
     border: 1px solid transparent;
-    border-radius: 4px;
+    border-radius: 50px;
```
Let's revert this back to 5px as pill shape does not fit within the UX.

MrRobotjs replied:
Reverting back to 4px, thinking about adding a css variable so it can be changed in a theme for users if wanted

therobbiedavis commented on UI/Web/src/app/sidenav/_components/side-nav-item/side-nav-item.component.scss:
```diff
@@ -22,21 +22,21 @@
     cursor: pointer;
-    font-size: 0.9rem;
+    font-size: 1.05rem;
+    font-family: sans-serif;
```
Can you explain why this change?

MrRobotjs replied:
The font size needed to be increased to match the increase of the icon size for the font family I was just testing if it looked better but after removing it now I realized it looks better without the change so I agree with that being reverted

therobbiedavis commented on UI/Web/src/theme/components/_sidenav.scss:
```diff
@@ -1,171 +1,236 @@
+            width: unset;
+            text-overflow: ellipsis;
+            display: block;
+            font-size: 13px;
```
Far too small and creates a ux discrepency between the home page sidebar and the preference sidebar. Let's adjust this to a manageable size for mobile as well.

MrRobotjs replied:
I’ll increase the font size for mobile, I took inspiration from the Plex settings side bar and thought that Kavita’s UX seems inspired by Plex so I tried to match that. I can increase the font size if needed along with mobile.

therobbiedavis commented on UI/Web/src/theme/components/_sidenav.scss:
```diff
@@ -1,171 +1,236 @@
+
+        &.active {
+          .active-highlight {
+            width: 4px;
```
I'm not sold on the active-highlight changes. Maybe workshop it a bit more, but we should try to keep a consistent ux and with this change the homepage sidebar and preference sidebar would have different behaviors.

MrRobotjs replied:
A rectangle active highlight didn’t make sense to me since it was floating and not flush with the left of the sidebar so again I took inspiration from the Plex settings sidebar and thought that made more sense and also since the settings sidebar and homepage have different uses (homepage tabs are flushed to the left of the sidebar and not indented, along with having card actions)

therobbiedavis reviewed (changes requested):
Let some comments. Overall I'm very open to some of these changes however one thing I want to stress is we want to keep the UX across the app cohesive. So if you're changing something in one place, make sure you're investigating to make sure it does not bring any experience discrepencies as a user.

MrRobotjs committed:
fixed input not being full width of search bar

input was maxed at 300px and search bar was longer than that so clicking anywhere passed the 300px would not do anything so making the input 100% width of the search bad fixed it

MrRobotjs committed:
added hover opacity on activity, settings, and user buttons

removed the box-shadow when :hover on the settings button and added instead a 0.8 opscity that changes to 1 when hovered

MrRobotjs committed:
removed font-family - unnecessary

MrRobotjs committed:
highlight changes to homepage sidebar

made it so the first highlight has a border radius to match with the side bar container
highlight radius will revert to 0px when in mobile

MrRobotjs committed:
redesigned preference side bar

reverted the highlight to be a rectangle
highlight will stay grey and visible until it is the active tab

MrRobotjs committed:
made side bar overlay full screen

when in mobile the overlay for the side bar had an offset for the nav bar, now the height is set for 100vh and z-index 1041 so it can be over everything that is not the side bar

MrRobotjs committed:
added padding left to side bar on mobile

MrRobotjs committed:
add side nav icon size variable and increased text color brightness

MrRobotjs committed:
clean up css

- moved majority of the css from preference.nav and side-nav-item to _sidenav.scss because there was a lot of conflicting and unnecessary css
- the homepage and preference sidebar css should be completely separate along with having the side-nav-item in their respective parent css file. there is no reason to separate the side-nav-item css since it's not being used anywhere else on the app

MrRobotjs committed:
active-highlight color change

visually nothing changed other than now it has opacity, so its transparent with the sidebar for when the background changes ie. opening a comic details page

MrRobotjs committed:
fixed text color

fixed so inputted text doesn't become white when search bar is out of focus (only happens if there are actual search results)

MrRobotjs committed:
changed search bar background color

changed from a solid color to a opaque color so it doesn't look out of place when going to a details page where the background color changes

MrRobotjs committed:
styling changes to subscribe button and active highlight fix

- subscribe now accurately matches sidebar nav items
- last active highlight now matches first active highlight

MrRobotjs committed:
fixed search bar overlay

covers entire screen now

MrRobotjs committed:
search bar fixes

fixed search bar x button alignment
fixed search bar input css conflicting with other input bars

MrRobotjs committed:
fixed selection bar from displaying incorrectly

- selection bar is now the width of the details page
- text size still needs to be lowered to fit on mobile devices correctly, but as of now it's functional

therobbiedavis committed:
Fixes zoom when clicking an input on mobile browsers

majora2007 commented:
I am taking a look at this PR now that canary has been merged into develop.

MrRobotjs committed:
set font-size: 16px to html

MrRobotjs committed:
convert logo to svg

- Added AI file of the logo
- Added SVG file of the logo
- replaced the logo in nav-header.component.html to an inline svg for higher quality

MrRobotjs committed:
Merge remote-tracking branch 'upstream/develop' into feature/qol-ui-changes

MrRobotjs committed:
px to rem / fixed mobile sidebar for preferences

- Fixed the preferences mobile sidebar from overflowing

MrRobotjs committed:
spacing improvements

- preference sidebar header margin-top removed
- settings header margin

MrRobotjs committed:
improved search bar responsiveness

- width increases more as screen gets smaller, 100% on most mobile devices

MrRobotjs committed:
sidebar active highlight width increased and text bold when active

MrRobotjs committed:
series page action buttons design improved

- "add want to read", "edit information", "more", "scrobbling" and "download" action buttons on the page of a series have been redesigned

majora2007 commented:
Sorry I merged a bunch of changes, once you update I can take a look.

MrRobotjs committed:
Merge remote-tracking branch 'upstream/develop' into feature/qol-ui-changes

majora2007 commented:
Once you wrap up my comments made in discord, let's rewrite the changelog note to match the expectation aka what changed for the user in plain text. 

If we want to keep the recap of the developer notes on the PR, we can make a comment with them.

MrRobotjs committed:
reverted logo icon svg to png image

MrRobotjs committed:
nav bar search background color reverted

- Background color of the search bar when focused is transparent with a border color

MrRobotjs committed:
matched main sidebar with preference sidebar for consistency

- increased sidebar active highlight width from 2px > 4px

MrRobotjs committed:
changed sidebar width and closed sidebar active color to white

MrRobotjs committed:
sidebar active color fixed

majora2007 commented on UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss:
```diff
@@ -18,73 +13,90 @@ form {
     box-shadow: none;
     cursor: text;
-    background-color: var(--input-bg-color);
+    background-color: rgb(255 255 255 / 12%);
```
We need a css variable for this.

majora2007 commented on UI/Web/src/app/app.component.scss:
```diff
@@ -14,6 +15,7 @@
     scrollbar-width: thin;
     mask-image: linear-gradient(to bottom, transparent, black 0%, black 95%, transparent 100%);
     -webkit-mask-image: linear-gradient(to bottom, transparent, black 0%, black 95%, transparent 100%);
+    //margin-top: 7px;
```
Remove dead code

majora2007 commented on UI/Web/src/app/nav/_components/grouped-typeahead/grouped-typeahead.component.scss:
```diff
@@ -18,73 +13,90 @@ form {
 
         &:empty {
-            padding-top: 6px !important;
+            /*padding-top: 6px !important;*/
```
Remove dead code

majora2007 commented on UI/Web/src/theme/components/_buttons.scss:
```diff
@@ -37,15 +37,28 @@
+  border-radius: 6px;
+
+  &:hover {
+    background-color: rgb(255 255 255 / 18%);
```
Need these to be a css property

majora2007 commented on UI/Web/src/theme/themes/dark.scss:
```diff
@@ -57,365 +57,319 @@
+  /* Alerts */
+  --alert-text-color: #fff3cd;
+  --alert-bg-color: transparent;
+  /* Checkboxes/Switch */
```
Please revert the indentation so it's easy to track changes. Bring the newlines back between the sections. If you're using auto formatters, please disable for this project.

majora2007 reviewed (changes requested):
Some comments and I noticed the icon alignment is no longer correct. 

Otherwise, the design changes I'm okay with for merging in. 

![image](https://github.com/user-attachments/assets/831921cb-b106-4bdc-b923-c85f4998bb6a)

majora2007 commented on UI/Web/src/theme/utilities/_global.scss:
```diff
@@ -1,5 +1,9 @@
 // Global styles for the site. Keep this as small as possible
 
+html {
+    font-size: 16px;
```
Make this a css variable

MrRobotjs committed:
added variables

- added html font size variable
- added btn actions border radius and hover bg variable
- reverted nav bar search input back to a variable for the bg color

MrRobotjs committed:
reverted brand logo back to png

- reverted the html from svg back to app-image for the brand logo

majora2007 commented:
![image](https://github.com/user-attachments/assets/b15de7eb-6666-4101-9205-509be09513f0)
The indicator is not looking like it is on the current
![image](https://github.com/user-attachments/assets/a4628580-e613-4a8f-b3fe-8494e20c8082)

Other than this and the merge conflicts, it's good to go. If you could try and wrap this today/tomorrow, that would be great. We are very close to a release boundary, I want to get this in for shakeout before then.

MrRobotjs committed:
fixed activity progress indicator in nav bar

changed top position from 30px to 24px

MrRobotjs committed:
Merge remote-tracking branch 'upstream/develop' into feature/qol-ui-changes

MrRobotjs commented:
> ![image](https://private-user-images.githubusercontent.com/735851/382878351-b15de7eb-6666-4101-9205-509be09513f0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MDMyMjksIm5iZiI6MTczMDkwMjkyOSwicGF0aCI6Ii83MzU4NTEvMzgyODc4MzUxLWIxNWRlN2ViLTY2NjYtNDEwMS05MjA1LTUwOWJlMDk1MTNmMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQxNDIyMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZTlmNGQ0YjEzNzc3YzAzZDFhYWRhZjk0ZDRiYjc4MDBjYzA0NjU1M2Y0NmY0YjE3MjVmNTI2MWUzZDY0MTg3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.O5vKuIlRK46P_y4RCslmlvKZGv358YgRpFp-VzdRfkY) The indicator is not looking like it is on the current ![image](https://private-user-images.githubusercontent.com/735851/382878453-a4628580-e613-4a8f-b3fe-8494e20c8082.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MDMyMjksIm5iZiI6MTczMDkwMjkyOSwicGF0aCI6Ii83MzU4NTEvMzgyODc4NDUzLWE0NjI4NTgwLWU2MTMtNGE4Zi1iM2ZlLTg0OTRlMjBjODA4Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEwNlQxNDIyMDlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZTBlMGE3NjNjYTM1MDZlNTk4YzU5NzYwMmFhZTIwZGUzOGRhZGQ5N2QxNWQ2NDI4Yzc3ZjRmMTNjYzI1NWM5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.x3csPS3OZ8zw3DZmxDsRI6wfgf01ota03M1SQnoZQxQ)
> 
> Other than this and the merge conflicts, it's good to go. If you could try and wrap this today/tomorrow, that would be great. We are very close to a release boundary, I want to get this in for shakeout before then.

Fixed

majora2007 commented:
I found a few bugs after merging your code into my branch. I'll ping you in discord if you can come and do a bit of polish afterwards.
